File: finstack-py/tests/golden/runners/pricing_deposit.py
"""Domain runner for `rates.deposit` golden fixtures."""

from __future__ import annotations

from tests.golden.schema import GoldenFixture

from .pricing_common import run_pricing_fixture


def run(fixture: GoldenFixture) -> dict[str, float]:
    """Run one deposit fixture through the Python bindings."""
    return run_pricing_fixture(fixture)

File: finstack/valuations/tests/golden/runners/pricing_deposit.rs
//! Domain runner for `rates.deposit` golden fixtures.

use crate::golden::runner::DomainRunner;
use crate::golden::schema::GoldenFixture;
use std::collections::BTreeMap;

/// Money-market deposit golden runner.
pub struct DepositRunner;

impl DomainRunner for DepositRunner {
    fn run(&self, fixture: &GoldenFixture) -> Result<BTreeMap<String, f64>, String> {
        crate::golden::runners::pricing_common::run_pricing_fixture(fixture)
    }
}
